plugins/csp-simple-objects/src/SimpleObject.cpp

```diff
@@ -0,0 +1,295 @@
+////////////////////////////////////////////////////////////////////////////////////////////////////
+//                               This file is part of CosmoScout VR                               //
+//      and may be used under the terms of the MIT license. See the LICENSE file for details.     //
+//                        Copyright: (c) 2022 German Aerospace Center (DLR)                       //
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#include "SimpleObject.hpp"
+
+#include <VistaKernel/GraphicsManager/VistaNodeBridge.h>
+#include <VistaKernel/VistaSystem.h>
+#include <VistaKernelOpenSGExt/VistaOpenSGMaterialTools.h>
+
+#include "../../../src/cs-core/Settings.hpp"
+#include "../../../src/cs-core/SolarSystem.hpp"
+#include "../../../src/cs-graphics/GltfLoader.hpp"
+//#include "../../../src/cs-graphics/internal/gltfmodel.hpp"
+#include "../../../src/cs-utils/convert.hpp"
+#include "../../../src/cs-utils/utils.hpp"
+
+#include <glm/gtc/type_ptr.hpp>
+#include <glm/gtx/quaternion.hpp>
+#include <utility>
+
+#include "logger.hpp"
+#include "utils.hpp"
+
+namespace csp::simpleobjects {
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+SimpleObject::SimpleObject(std::string const& name, Plugin::Settings::SimpleObject const& config,
+    VistaSceneGraph* sceneGraph, std::shared_ptr<cs::core::Settings> settings,
+    std::shared_ptr<cs::core::SolarSystem> solarSystem)
+    : mConfig(std::make_shared<Plugin::Settings::SimpleObject>(config))
+    , mSceneGraph(sceneGraph)
+    , mSettings(std::move(settings))
+    , mSolarSystem(std::move(solarSystem))
+    , mModel(std::make_shared<cs::graphics::GltfLoader>(mConfig->mModelFile, mConfig->mEnvironmentMap /*, true*/)) {
+
+
+  mAnchorObject = mSolarSystem->getBody(mConfig->mAnchorName);
+
+  mAnchor = std::unique_ptr<cs::scene::CelestialAnchorNode>(new cs::scene::CelestialAnchorNode(mSceneGraph->GetRoot(), mSceneGraph->GetNodeBridge(), name, 
+                                            mSettings->getAnchorCenter(mConfig->mAnchorName), mSettings->getAnchorFrame(mConfig->mAnchorName)));
+  mSettings->initAnchor(*mAnchor, mConfig->mAnchorName);
+  mAnchor->setAnchorScale(mConfig->mScale.get());
+
+// set initial position
+  auto lngLat = cs::utils::convert::toRadians(mConfig->mLngLat);
+  double height = (mAnchorObject->getHeight(lngLat) + mConfig->mElevation.get()) * mSettings->mGraphics.pHeightScale.get();
+  mAnchor->setAnchorPosition(cs::utils::convert::toCartesian(lngLat, mAnchorObject->getRadii(), height));
+  
+  // ground fixed rotation:
+  lastSurfaceNormal = cs::utils::convert::lngLatToNormal(cs::utils::convert::toRadians(mConfig->mLngLat));
+  qRot = utils::normalToRotation(lastSurfaceNormal);
+  mAnchor->setAnchorRotation(qRot);
+
+  mSolarSystem->registerAnchor(mAnchor);
+  mSceneGraph->GetRoot()->AddChild(mAnchor.get());
+  
+  //mModel->setEnableHDR(true);
```

Comment:
Could you please check the entire source file for comments which are not required anymore?